src/storage/mod.rs

```diff
@@ -171,32 +176,82 @@ impl fmt::Display for Command {
     }
 }
 
+impl Command {
+    pub fn readonly(&self) -> bool {
+        match *self {
+            Command::Get { .. } |
+            Command::BatchGet { .. } |
+            Command::Scan { .. } => true,
+            _ => false,
+        }
+    }
+}
+
 pub struct Storage {
     engine: Arc<Box<Engine>>,
-    sched: Option<Scheduler>,
+    schedch: SchedCh,
+    sched_handle: Option<thread::JoinHandle<()>>,
+    sched_event_loop: Option<EventLoop<Scheduler>>,
 }
 
 impl Storage {
-    pub fn from_engine(engine: Box<Engine>) -> Result<Storage> {
+    pub fn from_engine(engine: Box<Engine>, config: &Config) -> Result<Storage> {
         let engine = Arc::new(engine);
-        let sched = Scheduler::new(engine.clone());
+        let event_loop = try!(create_event_loop(config.sched_notify_capacity,
```

Comment:
I think leave the eventloop outside seems more clean and readable. Logically, `Storage` belongs to `EventLoop`.